ledfx/effects/audio.py

```diff
@@ -291,11 +311,25 @@ def activate(self):
             self.delay_queue = None
 
         def open_audio_stream(device_idx):
+            """
+            Opens an audio stream for the specified input device.
+            Parameters:
+            device_idx (int): The index of the input device to open the audio stream for.
+            Behavior:
+            - Detects if the device is a Windows WASAPI Loopback device and logs its name and channel count.
+            - If the device is a WEB AUDIO device, initializes a WebAudioStream and sets it as the active audio stream.
+            - For other devices, initializes an InputStream with the device's default sample rate and other parameters.
+            - Initializes a resampler with the "sinc_fastest" algorithm that downmixes the source to a single-channel.
+            - Logs the name of the opened audio source.
+            - Starts the audio stream and sets the audio stream active flag to True.
+            """
             device = input_devices[device_idx]
-            ch = 1
             if hostapis[device["hostapi"]]["name"] == "Windows WASAPI":
                 if "Loopback" in device["name"]:
-                    ch = 2
+                    ch = device["max_input_channels"]
```

Comment:
I would get rid of this ch var and just in line the device reference in the logging, setting it up as a var keeps making me think it is used elsewhere, but its not now.